frontend/features/page/sections/ContactsSection/ContactsSection.tsx

```diff
@@ -0,0 +1,753 @@
+  }
+}
+
+export function ContactsForm() {
+  const [state, dispatch] = React.useReducer(reducer, blankForm);
+
+  function validateErrors(values: FormState["values"]) {
+    const regex = /^[^@\s]+@[^@\s\.]+\.[^@\.\s]+$/;
+
+    const validationErrors: FormState["validationErrors"] = {
+      name: null,
+      email: null,
+      message: null,
+    };
+
+    if (values.name.trim().length === 0) {
+      validationErrors.name = "Please insert your name";
+    } else {
+      validationErrors.name = null;
+    }
+
+    if (!regex.test(state.values.email)) {
+      validationErrors.email = "Please insert a valid email";
+    } else {
+      validationErrors.email = null;
+    }
+    if (state.values.message.trim().length == 0) {
+      validationErrors.message = "Please insert a message";
+    } else {
+      validationErrors.message = null;
+    }
+    return validationErrors;
+  }
+
+  React.useEffect(() => {
+    if (state.status === FormStatus.Submitting) {
+      async function sendMessage() {
+        const input: CreateFormMessageInput = {
+          data: {
+            name: state.values.name,
+            email: state.values.email,
+            message: state.values.message,
+          },
+        };
+        const response = await fetchGraphQL<
+          InsertFormMessageMutation,
+          InsertFormMessageMutationVariables
+        >(InsertFormMessage, { input });
+
+        if (response.createFormMessage == null) {
+          dispatch({ type: FormActionType.Failed });
+        } else {
+          dispatch({ type: FormActionType.Success });
+        }
+      }
+      sendMessage();
+    }
+  }, [state.status, state.values]);
+
+  const onFieldChange = React.useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      const { name, value } = event.target;
+      dispatch({
+        type: FormActionType.UpdateField,
+        name,
+        value,
+      });
+    },
+    []
+  );
+
+  const onFormSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+
+    const validationErrors = validateErrors(state.values);
+
+    const hasNoErrors = Object.values(validationErrors).every(
+      (prop) => prop == null || prop === ""
+    );
+
+    if (hasNoErrors) {
+      dispatch({ type: FormActionType.Submit });
+    } else {
+      dispatch({
+        type: FormActionType.ValidationFailed,
+        validationErrors: validationErrors,
+      });
+    }
+  };
+
+  return (
+    <Box
+      flexWrap="wrap"
+      fontFamily="Roboto Mono"
+      id="contact"
+      mt={{
+        base: "20",
+        md: "0",
+      }}
+      mr={{
+        base: "-2.5",
+        md: "0",
+      }}
+      display="flex"
+      flexGrow={1}
+      pb="20"
+      flexDir={{
+        base: "column",
+        sm: "row",
+        md: "column",
+        lg: "row",
+      }}
+    >
+      {state.status === FormStatus.Submitted && (
+        <Box
+          display={
+            state.status === FormStatus.Submitted ? "inline-block" : "none"
```

Comment:
This condition seems useless, since we're already checking above that `state.status === FormStatus.Submitted`